src/net.cpp

```diff
@@ -589,14 +598,22 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
     while (nBytes > 0) {
         // absorb network data
         int handled = m_deserializer->Read(pch, nBytes);
-        if (handled < 0) return false;
+        if (handled < 0) {
+            // Serious header problem, disconnect from the peer.
+            return false;
+        }
 
         pch += handled;
         nBytes -= handled;
 
         if (m_deserializer->Complete()) {
             // decompose a transport agnostic CNetMessage from the deserializer
-            CNetMessage msg = m_deserializer->GetMessage(Params().MessageStart(), nTimeMicros);
+            Optional<CNetMessage> result = m_deserializer->GetMessage(nTimeMicros);
+            if (!result) {
+                // Message deserialization failed.  Drop the message but don't disconnect the peer.
+                continue;
+            }
+            CNetMessage msg = *result;
```

Comment:
style nit, feel free to ignore
```suggestion
            CNetMessage msg{*result};
```